app/questionnaire/rules/operations.py

```diff
@@ -49,9 +57,16 @@ class Operations:
     A class to group the operations
     """
 
-    def __init__(self, language: str) -> None:
+    def __init__(
+        self,
+        language: str,
+        schema: QuestionnaireSchema,
+        renderer: Optional["PlaceholderRenderer"] = None,
+    ) -> None:
         self._language = language
         self._locale = DEFAULT_LOCALE if language in {"en", "eo"} else language
+        self.renderer = renderer or None
```

Comment:
We should always pass in a renderer. Anything operations that depend on it will error at run time otherwise. A lot on placeholder parser shouldn't be optional either, but for anything new we added, we should always have them available, hence should pass it. We should accept that tests needs updating, as we shouldn't avoid because of that.